src/workspace-data/data-table/uri-viewer/UriPreview.js

```diff
@@ -22,23 +22,24 @@ const styles = {
   },
 };
 
-const isImage = ({ contentType, name }) => {
-  return /^image/.test(contentType) || /\.(?:(jpe?g|png|svg|bmp))$/.test(name);
+export const isImage = ({ contentType, name }) => {
+  return /^image/.test(contentType) || /\.(jpe?g|png|svg|bmp)$/.test(name);
 };
 
-const isText = ({ contentType, name }) => {
-  return /(?:(^text|application\/json))/.test(contentType) || /\.(?:(txt|[ct]sv|log|json))$/.test(name);
+export const isText = ({ contentType, name }) => {
+  return /(^text|application\/json)/.test(contentType) || /\.(txt|[ct]sv|log|json|fastq|fasta|fa|vcf|sam|bed|interval_list|gtf|md|html)$/.test(name);
 };
 
-const isBinary = ({ contentType, name }) => {
-  // excluding json and google's default types (i.e. wasn't set to anything else)
-  return (
-    /application(?!\/(json|octet-stream|x-www-form-urlencoded)$)/.test(contentType) || /(?:(\.(?:(ba[mi]|cra[mi]|pac|sa|bwt|gz))$|\.gz\.))/.test(name)
-  );
+export const isBinary = ({ contentType, name }) => {
+  return /application(?!\/(json|octet-stream|x-www-form-urlencoded)$)/.test(contentType) || /(\.(ba[mi]|cra[mi]|bcf|h5ad|pdf|gz)$|\.gz\.)/.test(name);
 };
 
-const isFilePreviewable = ({ size, ...metadata }) => {
-  return !isBinary(metadata) && (isText(metadata) || (isImage(metadata) && size <= 1e9));
+export const canRender = ({ contentType, name }) => {
+  return /(^text\/html|application\/pdf)/.test(contentType) || /\.(html|pdf)$/.test(name);
+};
```

Comment:
Allow rendering only for HTML and PDF files

Reply:
the `html` extension is also included in `isText`, and the `pdf` extension is included in `isBinary`. Consider removing them from isText/isBinary to avoid confusion; if you do this, I think you'll also have to modify `isFilePreviewable` to include an additional check against `canRender`.

Reply:
Also - I have some concern about rendering HTML here, due to the potential for it to execute malicious scripts. Those scripts are contained in an `<object>` tag so they are fairly well sandboxed, but it may still be an issue. My conservative suggestion is to ship the rest of this PR - minus html rendering - and raise the issue with appsec to get their input. Alternately you could hold the whole PR until you've had a chance to discuss with appsec.